### PR TITLE
Tweaks to t/one-core-peak.sh

### PR DESCRIPTION
**Pull request by ByteHamster**
A few little tweaks to t/one-core-peak.sh

- The [default](https://github.com/axboe/fio/blob/25425cb4a5531b1b3f26eba4e49866d944e0f1fb/t/io_uring.c#L97) number of threads in the benchmark is 1.
- When only one block device is passed, use one of the cores in its Numa domain instead of the first core.

CC @ErwanAliasr1

### CODE REVIEW

File: t/one-core-peak.sh [hunk 1]
```diff
@@ -1,10 +1,11 @@
 #!/bin/bash
 
 args=$*
+argcount=$#
 first_cores=""
 taskset_cores=""
 first_cores_count=0
-nb_threads=4 #default from the benchmark
+nb_threads=1
```
[the remark this answers is not in the document]
Comment (ByteHamster) [reply t/one-core-peak.sh:8]: Where does the value 4 come from? The `t/io_uring` test seems to [use 1](https://github.com/axboe/fio/blob/25425cb4a5531b1b3f26eba4e49866d944e0f1fb/t/io_uring.c#L97) and the [tweet](https://twitter.com/axboe/status/1435741761897254916) used 3.

[the remark this answers is not in the document]
Comment (ErwanAliasr1) [reply t/one-core-peak.sh:8]: yeah ... Dunno what I did here...

File: t/one-core-peak.sh [hunk 2]
```diff
@@ -161,19 +173,19 @@ show_device() {
...
+  info "system" "MEMORY: ${MEMORY_SPEED}"
+  info "system" "KERNEL: ${KERNEL}"
 }
 
 ### MAIN
 check_args $#
 check_root
 check_binary t/io_uring lscpu grep taskset cpupower awk tr xargs dmidecode
-detect_first_core
+detect_first_core $1
```
[the remark this answers is not in the document]
Comment (ErwanAliasr1) [reply t/one-core-peak.sh:188]: I have to test that too to confirm, @axboe would be interesting testing this on your high perf setup

File: t/one-core-peak.sh [hunk 3]
```diff
@@ -35,11 +36,22 @@ check_binary() {
...
+    cpu_to_search=$(cat ${pci_dir}/local_cpulist | cut -d"," -f 1 | cut -d"-" -f 1)
+  else
+    hint 'Passed multiple devices. Running on the first core.'
+  fi
+  core_to_run=$(lscpu  --all -pSOCKET,CORE,CPU | grep ",$cpu_to_search\$" | cut -d"," -f1-2)
+
   # Detect which logical cpus belongs to the first physical core
   # If Hyperthreading is enabled, two cores are returned
-  cpus=$(lscpu  --all -pSOCKET,CORE,CPU |grep "0,0")
+  cpus=$(lscpu  --all -pSOCKET,CORE,CPU | grep "$core_to_run")
```
[the remark this answers is not in the document]
Comment (ErwanAliasr1) [reply t/one-core-peak.sh:54]: I think that still match too much as two columns can match "x,"

File: t/one-core-peak.sh [hunk 4]
```diff
@@ -35,11 +36,22 @@ check_binary() {
...
 
-
 detect_first_core() {
+  cpu_to_search="0"
+  if [ "$argcount" -eq 1 ]; then
+    device_name=$(block_dev_name $1)
+    device_dir="/sys/block/${device_name}/device/"
+    pci_addr=$(cat ${device_dir}/address)
+    pci_dir="/sys/bus/pci/devices/${pci_addr}/"
+    cpu_to_search=$(cat ${pci_dir}/local_cpulist | cut -d"," -f 1 | cut -d"-" -f 1)
```
[the remark this answers is not in the document]
Comment (ErwanAliasr1) [reply t/one-core-peak.sh:46]: I have to test that too to confirm, @axboe would be interesting testing this on your high perf setup

File: t/one-core-peak.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (ErwanAliasr1) [reply t/one-core-peak.sh:188]: So I would probably extract the part detecting the cpulist of a given disk to a function, as we do it twice in this version of the tool, and pass the resulting cpu_core as parameter, not the diskname.

File: t/one-core-peak.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (ByteHamster) [reply t/one-core-peak.sh:54]: `$core_to_run` already contains two columns: `x,x`. Together with the comma, it becomes `x,x,` and therefore cannot match multiple columns

**Comment by ErwanAliasr1** [comment]
Can you rebase on master ?

**Comment by ErwanAliasr1** [comment]
Looks good to me.
It gives me a 7% boost on my setup.

**Comment by ErwanAliasr1** [comment]
@axboe LGTM

**Review by ErwanAliasr1 — APPROVED** [review]

**Pull request merged**